addons/agent_mirrorpeer_controller.go

```diff
@@ -50,6 +49,8 @@ type MirrorPeerReconciler struct {
 	SpokeClusterName     string
 	OdfOperatorNamespace string
 	Logger               *slog.Logger
+
+	testEnvFile string
```

Comment:
Instead of adding in the same struct with should embed this struct with another one specifically created for tests. Separating concerns for tests

Reply:
I am a bit confused with this pattern . Why are we adding changes to prod functions with test fields

Reply:
Whenever you are dealing with configurations, you want to be able to substitute as needed while testing. This seems to be the general pattern, used across kubernetes projects. See [example here](https://github.com/kubernetes-sigs/controller-runtime/blob/9f4e0f706b7094077734917db9f3655502a07a3e/pkg/manager/manager.go#L278-L283)

Technically, this functionality can be extended to production users as well. But, I do not see any use for the users yet. So I am calling it a test only feature.